md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/AppDeploymentQueueStatusAPI.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.wipro.ats.bdre.md.rest;
+
+import com.wipro.ats.bdre.md.api.base.MetadataAPIBase;
+import com.wipro.ats.bdre.md.beans.table.AppDeploymentQueueStatus;
+import com.wipro.ats.bdre.md.dao.AppDeploymentQueueStatusDAO;
+import com.wipro.ats.bdre.md.rest.util.BindingResultError;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Created by SU324335 on 3/8/2016.
+ */
+@Controller
+@RequestMapping("/adqstatus")
+public class AppDeploymentQueueStatusAPI extends MetadataAPIBase {
+
+    private static final Logger LOGGER = Logger.getLogger(AppDeploymentQueueStatusAPI.class);
+    private static final String RECORDWITHID = "Record with ID:";
+    @Autowired
+    AppDeploymentQueueStatusDAO appDeploymentQueueStatusDAO;
+
+    /**
+     * This method calls proc GetDeployStatus and fetches a record from DeployStatus table corresponding
+     * to deployStatusId passed.
+     *
+     * @param
+     * @return restWrapper It contains an instance of DeployStatus corresponding to deployStatusId passed.
+     */
+    @RequestMapping(value = "/{id}", method = RequestMethod.GET)
+    @ResponseBody public
+    RestWrapper get(
+            @PathVariable("id") Short appDeploymentStatusId, Principal principal
+    ) {
+
+        RestWrapper restWrapper = null;
+        try {
+            com.wipro.ats.bdre.md.dao.jpa.AppDeploymentQueueStatus jpaAdqStatus = appDeploymentQueueStatusDAO.get(appDeploymentStatusId.shortValue());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unboxing of "appDeploymentStatusId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS2153)